### PR TITLE
Team 5 player refactor

### PR DESCRIPTION
**Pull request by SIN-Willett**
# Description

The Player Factory has been significantly refactored, extracting the concept of a "PlayerFactoryFactory"

Fixes / Closes # (issue)
#410 

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

The pre-existing tests were used as regression tests.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by SIN-Willett** [comment]
@IanP96 hey I've killed off player selection and moved it into a `PlayerFactoryFactory` and put an abstract `PlayerFactory` in the `GameOptions`, could you double check this works with your code.

**Comment by SIN-Willett** [comment]
@Jas2310 and @Magnius001 I have straight up moved items and weapons into their own folders, and re-gutted weapon responsibilities out of the inventory, can you double check this works with your changes?

**Comment by SIN-Willett** [comment]
@vsinh14 I've pulled Pets out into their own kind of collectible, can you double check this works with your changes?

File: source/core/src/main/com/csse3200/game/components/player/PlayerFactoryFactory.java [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+            "player.json",
+            "player_2.json",
+            "player_3.json",
+            "player_4.json",
+            "bear.json",
+            "necromancer.json"
+    );
+
+    // List of all Options this Factory can create.
+    private final Map<String, PlayerConfig> options;
```
Comment (IanP96) [inline source/core/src/main/com/csse3200/game/components/player/PlayerFactoryFactory.java:35]: is this filename to config or player name to config?

**Review by IanP96 — APPROVED** [review]
looks good, just not really sure what `options` is. also there are like 5 conflicting files lol

**Comment by vsinh14** [comment]
There appear to be merge conflicts @SIN-Willett

**Review by Jas2310 — APPROVED** [review]
This looks good to me!

**Comment by vsinh14** [comment]
I resolved the merge conflicts but load does not follow convention with main so currently still breaks in this branch.
Please resolve this issue before the merge will be accepted

**Comment by Magnius001** [comment]
Load can be temporally fixed by checking for null when getting the multiplier for `difficulty` in `adjustForDifficulty()` of `PlayerConfig.java`.

Regarding the changes to weapons, everything looks good.

**Comment by IanP96** [comment]
> Load can be temporally fixed by checking for null when getting the multiplier for `difficulty` in `adjustForDifficulty()` of `PlayerConfig.java`.
> 
> Regarding the changes to weapons, everything looks good.

@SIN-Willett I think the best option would be to have a PlayerFactory.create that takes no arguments and does no difficulty scaling. Realistically there shouldn't be any need to reapply the difficulty, the effect of the difficulty selection is applied when you start the game that you saved if that makes sense

**Comment by vsinh14** [comment]
Loading still breaks the game

**Review by vsinh14 — APPROVED** [review]
All issues found through manual testing have been resolved 
No glaring faults in the code 
No javadoc errors

**Pull request merged**